games/xworld3d/maps/xworld3d_env.py

```diff
@@ -119,9 +122,19 @@ def delete_entity(self, loc=None, id=None):
         x = xe[0]
         self.entities.remove(x)
         self.entity_nums[x.type] -= 1
-        self.available_grids.append(x.loc)
+        ## agent has continuous loc, so we don't restore
+        if x.type != "agent":
+            self.available_grids.append(x.loc)
         self.changed = True
```

Comment:
I think we shouldn't allow deleting agent

Reply:
The teacher needs to move agent